File: Shared/PPPlugInInfoController.m
//
//  PPPlugInInfoController.m
//  PPMacho
//
//  Created by C.W. Betts on 12/31/12.
//
//

#import "PPPlugInInfoController.h"
#ifdef PLAYERPRO6
#import "PlayerPRO_6-Swift.h"
#else
#import "PlayerPRO_Player-Swift.h"
#endif

@interface PPPlugInInfoController ()
@property (strong) PPPlugInInfo *info;
@end

@implementation PPPlugInInfoController
@synthesize authorField;
@synthesize info;
@synthesize nameField;
@synthesize pluginImage;
@synthesize theCopyrightField;
@synthesize typeField;

- (id)initWithPlugInInfo:(PPPlugInInfo *)plugInfo;
{
	if (self = [self initWithWindowNibName:@"PPPlugInInfoController"]) {
		self.info = plugInfo;
	}
	return self;
}

- (IBAction)closeWindow:(id)sender {
	[NSApp stopModal];
	[[self window] close];
}

- (id)initWithWindow:(NSWindow *)window
{
	self = [super initWithWindow:window];
	if (self) {
		// Initialization code here.
	}
	
	return self;
}

- (void)windowDidLoad
{
	[super windowDidLoad];
	// Implement this method to handle any initialization after your window controller's window has been loaded from its nib file.
	[nameField setStringValue:self.info.plugName];
	[authorField setStringValue:self.info.authorName];
	[typeField setStringValue:self.info.plugType];
	[theCopyrightField setStringValue:self.info.plugCopyright];
	NSImage *plugImage = [[NSWorkspace sharedWorkspace] iconForFile:self.info.plugURL.path];
	[plugImage setSize:NSMakeSize(64, 64)];
	[pluginImage setImage:plugImage];
}

@end

File: Shared/PPPlugInInfoController.swift
//
//  PPPlugInInfoController.swift
//  PPMacho
//
//  Created by C.W. Betts on 8/11/14.
//
//

import Cocoa

class PPPlugInInfoController: NSWindowController {

	private var info: PPPlugInInfo!
	@IBOutlet var nameField: NSTextField!
	@IBOutlet var authorField: NSTextField!
	@IBOutlet var typeField: NSTextField!
	@IBOutlet var theCopyrightField: NSTextField!
	@IBOutlet var pluginImage: NSImageView!

	class func windowControllerFromInfo(plugInfo: PPPlugInInfo) -> Self {
		var toRet = self(windowNibName:"PPPlugInInfoController")
		toRet.info = plugInfo
		
		return toRet
	}
	
	//How do I even...
	/*
	class func getNibName() -> String {
		return "PPPlugInInfoController"
	}
	
	init(plugInInfo plugInfo: PPPlugInInfo) {
		info = plugInfo
		super.init(windowNibName: "PPPlugInInfoController")
	}*/
	
    override func windowDidLoad() {
		let plugImSize = NSMakeSize(64, 64)
        super.windowDidLoad()
    
        // Implement this method to handle any initialization after your window controller's window has been loaded from its nib file.
		
		nameField.stringValue = self.info.plugName
		authorField.stringValue = self.info.authorName
		typeField.stringValue = self.info.plugType
		theCopyrightField.stringValue = self.info.plugCopyright
		var plugImage = NSWorkspace.sharedWorkspace().iconForFile(info.plugURL.path)
		plugImage.size = plugImSize
		pluginImage.image = plugImage
    }
	
	@IBAction func closeWindow(sender: AnyObject?) {
		NSApplication.sharedApplication().stopModal()
		self.window.close()
	}
}
